File: OrchidCore/src/main/java/com/eden/orchid/api/options/globals/IndexGlobal.java
package com.eden.orchid.api.options.globals;

import com.eden.orchid.api.OrchidContext;
import com.eden.orchid.api.indexing.OrchidIndex;
import com.eden.orchid.api.options.TemplateGlobal;

public class IndexGlobal implements TemplateGlobal<OrchidIndex> {

    @Override
    public String key() {
        return "index";
    }

    @Override
    public OrchidIndex get(OrchidContext context) {
        return context.getIndex();
    }
}

File: OrchidCore/src/main/kotlin/com/eden/orchid/api/options/globals/IndexGlobal.kt
package com.eden.orchid.api.options.globals

import com.eden.orchid.api.OrchidContext
import com.eden.orchid.api.indexing.OrchidIndex
import com.eden.orchid.api.options.TemplateGlobal
import com.eden.orchid.api.theme.pages.OrchidPage

class IndexGlobal : TemplateGlobal<OrchidIndex?> {
    override fun key() = "index"
    override fun get(context: OrchidContext, page: OrchidPage?): OrchidIndex? = context.index
}
